### DISCUSSION TITLE
How does Django Unicorn compare with  or borrow from Sock Puppet

### DISCUSSION THREAD

**Discussion by tkrugg** [django-commons/django-unicorn#171] [Q&A] [answered]
I'm very excited about Django Unicorn.

Recently, I stumbled on an older project https://github.com/jonathan-s/django-sockpuppet. I'd love to know if the creators were aware of this project and in what way they do (or will) differentiate.

**Comment by adamghill** [comment, marked as answer]
Hey @tkrugg, thanks for the question! Yep, I call out `sockpuppet` (and some other related projects) at https://www.django-unicorn.com/docs/#full-stack-framework-python-packages.

The major difference (as I see it) is that `Unicorn` is designed to be as easy as possible to add to a Django project which means that it _only_ uses AJAX calls to work its magic. Most of the other projects I know of for Django all use websockets (which requires Django Channels, redis, or some other pub-sub technology). The trade-off is that an AJAX call _will_ be slightly slower than a persistent websocket connection. However, I've spent some time optimizing the size of the network call and provide some ways to reduce the size of the network packet. I'm planning on continuing to focus on performance so that `Unicorn` will be as fast as possible.

https://www.django-unicorn.com/docs/faq/#what-is-the-difference-between-unicorn-and-lighter-front-end-frameworks-like-htmx-or-alpine-js talks specifically about `HTMX` and `alpine.js`, but it gets to one of the other goals for `Unicorn` which is to integrate tightly with Django to make it _feel_ like it's a part of the framework.

Hopefully that all makes sense, but let me know if you have any other questions or comments!

**Reply by tkrugg** [reply]
It does make sense, thank you very much!
